src/js/line.js

```diff
@@ -0,0 +1,493 @@
+        },
+        modelListeners: {
+            dataSet: {
+                funcName: "{that}.draw",
+                excludeSource: "init"
+            }
+        },
+        modelRelay: {
+            source: "{that}.model.dataSet",
+            target: "{that}.model.wrappedDataSet",
+            singleTransform: {
+                type: "fluid.transforms.free",
+                args: ["{that}.model.dataSet"],
+                func: "floe.chartAuthoring.lineChart.chart.wrapSingleDataSet"
+            }
+        },
+        invokers: {
+            draw: {
+                funcName: "floe.chartAuthoring.lineChart.chart.draw",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    // Test if a dataset is multi
+    floe.chartAuthoring.lineChart.chart.isMultiDataSet = function (dataSet) {
+        if (dataSet[0].id !== undefined && dataSet[0].data !== undefined) {
+            return true;
+        } else {
+            return false;
+        }
+    };
+
+    // Wrap a non-multi dataset (a simple array without ID keys) so we can
+    // process it with the same function used for multi datasets, and
+    // create a default id for it for object constancy
+    floe.chartAuthoring.lineChart.chart.wrapSingleDataSet = function (dataSet) {
+        var wrapped = [
+            {id: "dataSet",
+            data: dataSet }
+        ];
+        return (floe.chartAuthoring.lineChart.chart.isMultiDataSet(dataSet)) ? dataSet : wrapped;
+    };
+
+    floe.chartAuthoring.lineChart.chart.draw = function (that) {
+
+        var shouldAddArea = that.options.lineOptions.addArea,
+            shouldAddPoints = that.options.lineOptions.addPoints;
+
+        that.yScale = floe.chartAuthoring.lineChart.chart.getYScale(that);
+
+        that.xScale = floe.chartAuthoring.lineChart.chart.getXScale(that);
+
+        that.yAxis = floe.chartAuthoring.lineChart.chart.getYAxis(that);
+
+        that.xAxis = floe.chartAuthoring.lineChart.chart.getXAxis(that);
+
+        that.line = floe.chartAuthoring.lineChart.chart.getLineGenerator(that);
+
+        that.area = floe.chartAuthoring.lineChart.chart.getAreaGenerator(that);
+
+        floe.chartAuthoring.lineChart.chart.addYAxis(that);
+
+        floe.chartAuthoring.lineChart.chart.addXAxis(that);
+
+        floe.chartAuthoring.lineChart.chart.addChartLine(that);
+
+        if (shouldAddArea) {
+            floe.chartAuthoring.lineChart.chart.addArea(that);
+        }
+
+        if (shouldAddPoints) {
+            floe.chartAuthoring.lineChart.chart.addPoints(that);
+        }
+
+        that.events.onChartRedrawn.fire();
+
+    };
+
+    floe.chartAuthoring.lineChart.chart.manageAxis = function (that, axisSelector, axisClass, axisTransform, axisFunction) {
+
+        var transitionLength = that.options.lineOptions.transitionLength;
+
+        var axisExists = (that.locate(axisSelector).length > 0) ? true : false;
+
+        // Append the axis if it's not drawn yet
+        if (!axisExists) {
```

Comment:
Not 100% sure I understand the comment, but I revised to the following:

```
        var noAxisExists = (that.locate(axisSelector).length > 0) ? false : true;

        if (noAxisExists) {
            // Append the axis if it's not drawn yet
            that.svg.append("g")
                .attr({
                    "transform": axisTransform,
                    "class": axisClass
                })
                .call(axisFunction);
        } else {
            // Transition the axis if it's already drawn
            that.svg.select("." + axisClass)
                .transition()
                .duration(transitionLength)
                .call(axisFunction);
        }
```

I think this is more natural in that it doesn't rely on a logical inversion for the `if` test.